### DISCUSSION TITLE
Jest encountered an unexpected token error

### DISCUSSION THREAD

**Discussion by saraschoonover** [nss-evening-cohort-14/evening-client-side#32] [Help Me!] [answered]
Have you searched the other help tickets? Yes
Describe your issue:

When I attempted to test my code  after multiple successes with the (npm test main.test.js) command, I got an error message:

 FAIL  ./main.test.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript."

and:

" Details:

    SyntaxError: /Users/saraschoonover/workspace/foundations/exercises/js-practice-1-SaraSchoonover/main.js: Unexpected token (205:0)

      203 |   getTriangleArea,
      204 | };
    > 205 | 
"
What have you googled? Didn't think this was googlable. 

What have you tried?  Made sure command was typed properly and reread the error message.

Copy/paste link TO THE FILE on your branch or in your repo: this is in my VS Code terminal.

**Comment by drteresavasquez** [comment]
Is there an extra space or something on line 205?

**Reply by saraschoonover** [reply]
There is no space there. Were those module.exports there the entire time? Or were they added while we were working on them? I  was coasting on through until it started telling me there was an error on the module.exports. It's possible I accidentally typed something down there, but I can't figure it out.

**Reply by drteresavasquez** [reply]
There has to be something there like an extra line. It could be because of the settings we did in vscode as well but it doesn’t seem like anyone else is having that issue so I’m not sure that is it. Make sure to backspace all the way up to the ; and then save and try to run the test.

**Reply by drteresavasquez** [reply]
Also, can you paste the whole code block starting from the exports?

**Reply by saraschoonover** [reply]
// Do not modify code below this line.
// --------------------------------

module.exports = {
  newString,
  newNum,
  newBool,
  newSubtract,
  newMultiply,
  newModulo,
  returnString,
  areSameLength,
  areEqual,
  lessThanNinety,
  greaterThanFifty,
  add,
  subtract,
  divide,
  multiply,
  getRemainder,
  isEven,
  isOdd,
  square,
  cube,
  raiseToPower,
  roundNumber,
  roundUp,
  addExclamationPoint,
  combineNames,
  getGreeting,
  getRectangleArea,
  getTriangleArea,
};

**Reply by drteresavasquez** [reply]
Try this. Go back to your main branch and try to run the test without making any updates.

**Comment by drteresavasquez** [comment]
Also, the same as I commented on the last ticket ALWAYS google the error. Even if you don’t think it will return results.

**Reply by saraschoonover** [reply]
I believe mine started out with 'master' for this. is master what I should go to instead?

**Reply by drteresavasquez** [reply]
Yes

**Reply by saraschoonover** [reply]
I did, ran the prompt, and got the same response

**Reply by drteresavasquez** [reply]
You have no changes on master right? If not, delete the node_modules directory and package.lock.JSON file and then run npm install again. Then see if the tests work. If they don’t we will have to look next week when we are in class. 

Also make sure that all of your brackets and elements are closed in your main.js file. It’s possible you missed something above in the file as well.

**Comment by drteresavasquez** [comment]
@SaraSchoonover watch this video to help you complete the assignment without needing the code on your local machine. It is a little more cumbersome, but at least it gets you an option to complete the assignment. We can discuss your issue and take a look on Tuesday.

https://vimeo.com/501565182/59d6c13c20

**Comment by Gabrielle-Tobermann** [comment]
@drteresavasquez just thought you should know, I ran into the same issue as Sara when doing the assignment directly on Github. I checked to make sure no curly braces were missing, no extra lines etc.  Everything seems to be correct.

**Reply by drteresavasquez** [reply]
Try removing the 'use strict' at the top of the document. That may help.

**Reply by drteresavasquez** [reply]
Your formatting has to be correct as well. I have run this several times myself and it runs fine. Make sure your indentions and code are correct for every function as well as is pointed out here: https://github.com/nss-evening-cohort-14/evening-client-side/discussions/34

**Comment by drteresavasquez** [comment, marked as answer]
@Gabrielle-Tobermann your function 17 does not have a closing bracket. This is why you are getting this error:

```
// 17
function isEven(num) {
  // return true if num is even
  // otherwise return false
  // code here
  if (num % 2 === 0) {
    return true;
  } else {
    return false;
  }
```

**Reply by Gabrielle-Tobermann** [reply]
Thank you!

**Comment by drteresavasquez** [comment]
@SaraSchoonover it looks like you were able to resolve your issue. Please let us know what you did to get it fixed ASAP. Thank you!

**Reply by saraschoonover** [reply]
I had an open bracket somewhere and just needed to find it and close it. Thank you for you help.
